### DISCUSSION TITLE
Simple block example returning nil

### DISCUSSION THREAD

**Discussion by dylan-hoefsloot** [prawnpdf/prawn#1289] [Q&A] [answered]
I tried to use the simple block example from the documentation
```ruby
document = Prawn::Document.generate("hello.pdf") do
  text "Hello World!"
end
```
and I expected document to contain a `Prawn::Document` but it is `nil`

```ruby
document = Prawn::Document.new
```
will return a `Prawn::Document` as I expected 

---

Is this expected behaviour?

Am I doing something wrong here?

**Comment by dylan-hoefsloot** [comment]
I have found that is does correctly generate the file to `hello.pdf` and if this is expected behaviour so be it, however It is a little bit confusing as the [documentation](https://prawnpdf.org/api-docs/2.3.0/Prawn/Document.html#generate-class_method) says the `#generate` method returns `Object`

**Comment by gettalong** [comment, marked as answer]
As you can see in the documentation you referenced, the return value is never used in the examples. Also, it doesn't say anywhere that the document instance is returned.

The "Object" that is mentioned on the method definition line  in the documentation as return value is probably just some Yard thing in that they always have to display something there.
